### PR TITLE
[android/mxnet] Add MXNet subplugin support @open sesame 05/27 16:58

### PR DESCRIPTION
**Pull request by makesource**
We newly added MXNet subplugin support for tensor_filter in https://github.com/nnstreamer/nnstreamer/pull/3742.
To run MXNet subplugin in android application, we added related implementation in api scripts.
We prepared prebuilt MXNet 1.9.1 libraries (built by NDK) and linked them at `Android-mxnet.mk`, `Android-mxnet-prebuilt.mk`.

You can build nnstreamer.aar with MXNet by following way.
- set nnstreamer to https://github.com/nnstreamer/nnstreamer/pull/3742.
- run `bash java/build-nnstreamer-android.sh --enable_mxnet=yes`

### CODE REVIEW

File: c/include/ml-api-common.h [hunk 1]
```diff
@@ -69,6 +69,7 @@ typedef enum {
   ML_NNFW_TYPE_NNTR_INF = 12,         /**< Inference supported from NNTrainer, SR On-device Training Framework (Since 6.5) */
   ML_NNFW_TYPE_VD_AIFW = 13,          /**< Inference framework for Samsung Tizen TV (Since 6.5) */
   ML_NNFW_TYPE_TRIX_ENGINE = 14,      /**< TRIxENGINE accesses TRIV/TRIA NPU low-level drivers directly (.tvn). (Since 6.5) You may need to use high-level drivers wrapping this low-level driver in some devices: e.g., AIFW */
+  ML_NNFW_TYPE_MXNET = 15,            /**< Apache MXNet (.params, .json). */
```
Comment (anyj0527) [inline c/include/ml-api-common.h:72]: Please set to `0x2002` and move to under the `ML_NNFW_TYPE_SNAP`.  And add comment sth like "only for Android"

File: java/build-nnstreamer-android.sh [hunk 2]
```diff
@@ -423,6 +437,10 @@ if [[ $enable_pytorch == "yes" ]]; then
     wget --directory-prefix=./$build_dir/external https://github.com/nnstreamer/nnstreamer-android-resource/raw/master/external/pytorch-$pytorch_ver.tar.xz
 fi
 
+if [[ $enable_mxnet == "yes" ]]; then
+    wget --load-cookies ~/cookies.txt "https://docs.google.com/uc?export=download&confirm=$(wget --quiet --save-cookies ~/cookies.txt --keep-session-cookies --no-check-certificate 'https://docs.google.com/uc?export=download&id=1BVYRmkVsTJ8SMZr0qRbYVQvFyiGPAgpF' -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\1\n/p')&id=1BVYRmkVsTJ8SMZr0qRbYVQvFyiGPAgpF" -O ./$build_dir/external/mxnet-1.9.1.tar.xz && rm -rf ~/cookies.txt
```
[the remark this answers is not in the document]
Comment (jaeyun-jung) [reply java/build-nnstreamer-android.sh:441]: Let's consider multiple files zipped under 30 MB in resource repo.
And, when uploading these files, please consider adding readme or announce about mxnet prebuilt files.
This should include the information about MXnet: link to mxnet website, version, commit ID or version tag, build option (or how to generate mxnet.so), ...

File: c/include/ml-api-common.h (hunk 1 above)
Comment (jaeyun-jung) [reply c/include/ml-api-common.h:72]: To reviewers
This change requires new ACR for Tizen release. There are two options to fix it, 
1. just do ACR process (mxnet only for android)
2. internal nnfw-type control with flag (0x2000 is nnfw for android)

File: java/build-nnstreamer-android.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (anyj0527) [reply java/build-nnstreamer-android.sh:441]: How about split the zipped file and upload those split files like in https://github.com/nnstreamer/nnstreamer-android-resource/pull/30 ?

File: c/include/ml-api-common.h (hunk 1 above)
Comment (myungjoo) [reply c/include/ml-api-common.h:72]: Option 1.

**Review by jaeyun-jung — APPROVED** [review]
LGTM. Please upload prebuilt library into android-res repogitory.

File: java/android/nnstreamer/src/main/jni/nnstreamer-native-api.c [hunk 3]
```diff
@@ -810,6 +816,9 @@ nnstreamer_native_initialize (JNIEnv * env, jobject context)
 #if defined (ENABLE_PYTORCH)
     init_filter_torch ();
 #endif
+# if defined (ENABLE_MXNET)
```
Comment (jaeyun-jung) [inline java/android/nnstreamer/src/main/jni/nnstreamer-native-api.c:819]: plz fix indent
```
#if defined (ENABLE_MXNET)
    init_filter_mxnet ();
#endif
```

File: java/build-nnstreamer-android.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (makesource) [reply java/build-nnstreamer-android.sh:441]: I uploaded pre-built mxnet library with proper documentation.
https://github.com/nnstreamer/nnstreamer-android-resource/pull/31/files

[the remark this answers is not in the document]
Comment (jaeyun-jung) [reply java/build-nnstreamer-android.sh:441]: @makesource It was merged, please update build-nnstreamer-android.sh and rebase this PR :)

**Review by anyj0527 — APPROVED** [review]
Cool! Thanks for your contribution. LGTM✔

**Pull request merged**